src/app/AttributeAccessInterceptor.h

```diff
@@ -0,0 +1,91 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/ClusterInfo.h>
+#include <app/util/basic-types.h>
+#include <lib/core/CHIPTLV.h>
+#include <lib/core/Optional.h>
+
+/**
+ * Callback class that clusters can implement in order to interpose
+ * custom attribute-handling logic.  An interceptor must intercept access for
+ * some specific cluster.  It may intercept access for one specific endpoint or
+ * for all endpoints.
+ */
+namespace chip {
+namespace app {
+
+class AttributeAccessInterceptor
```

Comment:
The act of interception is a specific policy decision. We shouldn't bake the policy detail in the naming here.

```suggestion
class AttributeAccessInterface
```

Reply:
Well, the point of this interface is to allow you to intercept reads/writes.  This is not the entire interface for attribute access; it's a way to interpose something in front of the "normal" codepath.

Reply:
OK, sold.  Will make this name change and do some other renaming of things.